### ISSUE TITLE
Accessing ClassVar/@staticmethod/@classmethod from instance should trigger error

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#196]
**Is your feature request related to a problem? Please describe.**
Current version of pyright (1.0.55) won't report error when accessing `ClassVar`, `@staticmethod ` and `@classmethod` from instance, and IntelliSense will also suggest them.

![image](https://user-images.githubusercontent.com/3479453/62940930-4b5dd580-be07-11e9-93ac-e0224a84c3cf.png)

**Describe the solution you'd like**
Though it's valid in runtime, accessing `ClassVar`, `@staticmethod` `@classmethod` from instance should trigger error, and pyright shoundn't suggest them as instance member in IntelliSense.

_I don't know will this break anything for some use case, so maybe there should be a config option for this?_

```python
from typing import ClassVar

class A:
    some_class_var: ClassVar[str]

    @classmethod
    def some_class_method(cls):
        pass

    @staticmethod
    def some_static_method():
        pass

a = A()
a.some_class_var # <- should report error
a.some_class_method # <- should report error
a.some_static_method # <- should report error
```

**Comment by erictraut** [comment]
I don't think these should generate errors. It's perfectly legit to access class variables, static methods and class methods from a class instance in python.

**Comment by giyyapan** [comment]
I see.

I understand it's legit, but not all legit thing in python are considered "good design pattern" or "type friendly".

One of the goal of static checker is to find out potential error, and accessing ClassVar/static member from instance instead of class smells like one. (Because the developer may thinking there dealing with some instance level context when doing so).

So I still think this could be an optional check for this.

![image](https://user-images.githubusercontent.com/3479453/62989624-53eff380-be7b-11e9-8b58-a1e05039282e.png)
_The official doc also gives an example of these kind of usage, and I simply don't know why should I ever use "ClassVar" otherwise_

Anyway, I you still think this is not a feature that pyright would want to add, feel free to close this issue. :smile:

**Comment by erictraut** [comment]
I don't see why this would be unsafe or undesirable. Accessing class variables from an instance is a key design point in Python — and a number of other OOP languages.

Note that the converse (attempting to access an instance variable from a class) is a bug, and pyright properly reports this as an error. For example...

```
class A(object):
    def __init__(self):
        self.X = 0

x = A.X
```

**Comment by giyyapan** [comment]
Understood.
Thanks for your explaination!
